### PR TITLE
fix(linux): can't get local ip address with libc::ENETUNREACH error

### PR DESCRIPTION
**Pull request by jsydliuqing**
Fix(linux): can not get local ip address when the network namespace without default route
#135

### CODE REVIEW

File: src/linux.rs [hunk 1]
```diff
@@ -158,6 +158,16 @@ fn local_ip_impl(family: RtAddrFamily) -> Result<IpAddr, Error> {
     let mut netlink_socket = NlSocketHandle::connect(NlFamily::Route, None, &[])
         .map_err(|err| Error::StrategyError(err.to_string()))?;
 
+    match local_ip_impl_route(family, &mut netlink_socket) {
+        Ok(ip_addr) => Ok(ip_addr),
+        Err(_error) => local_ip_impl_addr(family, &mut netlink_socket),
```
Comment (EstebanBorai) [inline src/linux.rs:163]: Can we check here for the error to be "LocalIpAddressNotFound"? That way we have a controlled flow for the fallback, allowing us to reduce ambiguity. 

```suggestion
        Err(error) if error == Error::LocalIpAddressNotFound => local_ip_impl_addr(family, &mut netlink_socket),
        Err(e) => Err(e)
```

Comment (jsydliuqing) [reply src/linux.rs]: Yes, that's better!

**Review by EstebanBorai — APPROVED** [review]
LGTM! Great work @jsydliuqing!

**Comment by EstebanBorai** [comment]
@jsydliuqing can you please run `cargo clippy` to fix fmt issues please?

**Review by EstebanBorai — APPROVED** [review]
Great @jsydliuqing! Thanks so much!

**Comment by jsydliuqing** [comment]
Hi @EstebanBorai !
When will the next new version be released?  I will use it in my project.

**Comment by EstebanBorai** [comment]
> Hi @EstebanBorai !
> 
> When will the next new version be released?  I will use it in my project.

Hey @jsydliuqing! I can do it in a couple hours!

**Comment by EstebanBorai** [comment]
Hi @jsydliuqing! Here is the version containing your PR changes: https://crates.io/crates/local-ip-address

**Pull request merged**